src/components/PopoverWithoutOverlay/index.js

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import {View} from 'react-native';
+import {SafeAreaInsetsContext} from 'react-native-safe-area-context';
+import {PopoverContext} from '../PopoverProvider';
+import * as Modal from '../../libs/actions/Modal';
+import {propTypes, defaultProps} from '../Popover/popoverPropTypes';
+import styles from '../../styles/styles';
+import * as StyleUtils from '../../styles/StyleUtils';
+import getModalStyles from '../../styles/getModalStyles';
+import withWindowDimensions from '../withWindowDimensions';
+
+function Popover(props) {
+    const ref = React.useRef(null);
+    const {onOpen, close} = React.useContext(PopoverContext);
+    const {modalStyle, modalContainerStyle, shouldAddTopSafeAreaMargin, shouldAddBottomSafeAreaMargin, shouldAddTopSafeAreaPadding, shouldAddBottomSafeAreaPadding} = getModalStyles(
+        'popover',
+        {
+            windowWidth: props.windowWidth,
+            windowHeight: props.windowHeight,
+            isSmallScreenWidth: false,
+        },
+        props.anchorPosition,
+        props.innerContainerStyle,
+        props.outerStyle,
+    );
+
+    React.useEffect(() => {
+        if (props.isVisible) {
+            props.onModalShow();
+            onOpen({
+                ref,
+                close: props.onClose,
+                anchorRef: props.anchorRef,
+            });
+        } else {
+            props.onModalHide();
+            close(props.anchorRef);
+        }
+        Modal.willAlertModalBecomeVisible(props.isVisible);
+        Modal.setCloseModal(props.isVisible ? () => props.onClose(props.anchorRef) : null);
```

Comment:
✋ Coming from https://github.com/Expensify/App/issues/27244

There will be a chance of race condition when the new popover `setCloseModal` will be overwritten by the previous popover, which `isVisible` property is changed into false (about to be closed) and make one of the pop over stuck.